keysign/util.py

```diff
@@ -29,13 +30,14 @@
 
 
 def mac_generate(key, data):
-    mac = hmac.new(key, data).hexdigest().upper()
+    mac = hmac.new(key, data, hashlib.sha256).hexdigest().upper()
     log.info("MAC of %r is %r", data[:20], mac[:20])
-    return mac
+    return mac[:20]
+
 
 def mac_verify(key, data, mac):
     computed_mac = mac_generate(key, data)
-    result = hmac.compare_digest(mac.upper(), computed_mac.upper())
```

Comment:
Hrm.
>  But if we truncate the mac_generate, then when we scan a qr we only have 20 chars of the hash and we need to set a limit to the compare or it will fails.
I think I haven't understood why it would fail.

Let's assume we pass a 20 byte long mac to `mac_verify`. It will call `mac_generate` which itself will generate a 20 byte long MAC. Then it should be verify correctly, no?